### ISSUE TITLE
Support arbitrary user data for a radish run

### ISSUE THREAD

**Issue by timofurrer** [radish-bdd/radish#124]
Instead of having the possibility to specify a `profile` using `--profile` the user should have the option to pass arbitrary user data into a radish run.

For that `-u` / `--user-data` should be introduced which can be specified multiple times and takes a `key=value` pair. This should be represented as a `dict` and assigned to the `world.config` object.

**Comment by rscrimojr** [comment]
@timofurrer This feature may be really useful to me.  I will see if I have time to add this.

**Comment by rscrimojr** [comment]
@timofurrer Do you want this in the `core/main` or as an `extension`?  Also, are you saying to remove `-p/--profile` option or is this in addition to the `-p/--profile` argument? It sounds like an addition. Thx

**Comment by rscrimojr** [comment]
@timofurrer I have an extension ready to PR.  Let me know if this is okay to proceed.

**Comment by timofurrer** [comment]
> Do you want this in the core/main or as an extension? My first thought is extension.

I think an extension is fine for now if you have a PR ready.

> Also, are you saying to remove `-p/--profile` option or is this in addition to the `-p/--profile` argument?

Actually I was planning to remove the profile option since it would be kind of obsolete - but we can keep it for one or two releases and warn to user about it's deprecation.

> @timofurrer I have an extension ready to PR. Let me know if this is okay to proceed.

Sounds great! Let's submit it then :tada:

**Comment by rscrimojr** [comment]
@timofurrer Do you want me to add the PendingDeprecationWarning in main.py in main():
```
    warnings.simplefilter('always', PendingDeprecationWarning)
 
    ...
    
    core = Core()

    if world.config.profile:
        msg = "-p/--profile will be removed in a future version.  Please use -u/--user-data instead."
        warnings.warn(msg, PendingDeprecationWarning)
    ...
```

Or do you want to do something different?  I can PR the extension without the PendingDeprecationWarning. Let me know what you want me to do.

**Comment by timofurrer** [comment]
No, using `warnings` and the filter sounds good to me.
Why using `PendingDep..` and not `DeprecationWarning` directly?

Just submit the PR and will review together and find out what works best.

**Comment by rscrimojr** [comment]
I guess since we are deprecating the profile argument in the next version, it makes sense to use DeprecationWarning instead.  I will change it and PR.  I was thinking you were going to deprecate in the future. Thx.

**Comment by rscrimojr** [comment]
Not sure how to create the tests for this enhancement.  Where should these tests be added? or do you want to take care of the tests?

**Comment by timofurrer** [comment]
Closed with #127. Released with **v0.6.2**
